homeassistant/components/watergate/sensor.py

```diff
@@ -0,0 +1,249 @@
+        translation_key="wifi_uptime",
+        key="wifi_uptime",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+        device_class=SensorDeviceClass.DURATION,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.mqtt_uptime if data.networking else None,
+        translation_key="mqtt_uptime",
+        key="mqtt_uptime",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+        device_class=SensorDeviceClass.DURATION,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.telemetry.water_temperature
+        if data.telemetry
+        else None,
+        translation_key="water_temperature",
+        key="water_temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.telemetry.pressure if data.telemetry else None,
+        translation_key="water_pressure",
+        key="water_pressure",
+        native_unit_of_measurement=UnitOfPressure.MBAR,
+        device_class=SensorDeviceClass.PRESSURE,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: (data.telemetry.flow / 1000)
+        if (data.telemetry and data.telemetry.flow is not None)
+        else None,
+        translation_key="water_flow_rate",
+        key="water_flow_rate",
+        native_unit_of_measurement=UnitOfVolumeFlowRate.LITERS_PER_MINUTE,
+        device_class=SensorDeviceClass.VOLUME_FLOW_RATE,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.state.uptime if data.state else None,
+        translation_key="uptime",
+        key="uptime",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        device_class=SensorDeviceClass.DURATION,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.state.power_supply if data.state else None,
+        translation_key="power_supply",
+        key="power_supply",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: WatergateConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up all entries for Watergate Platform."""
+
+    coordinator = config_entry.runtime_data
+
+    entities: list[SensorEntity] = [
+        SonicSensor(coordinator, description) for description in DESCRIPTIONS
+    ]
+
+    entities.extend([AutoShutOffEventSensor(coordinator)])
+
+    async_add_entities(entities, True)
```

Comment:
Why do you set this to `True`?
I think this would cause the coordinator to run a second time.

Reply:
Correct! Thanks.